api/app/GraphQL/Mutations/UpdatePoolCandidateStatus.php

```diff
@@ -12,10 +12,17 @@
     public function __invoke(null $_, array $args)
     {
         $candidate = PoolCandidate::findOrFail($args['id']);
+        $values = [];
 
         if (isset($args['pool_candidate_status']) && $args['pool_candidate_status'] !== $candidate->pool_candidate_status) {
             $status = $args['pool_candidate_status'];
 
+            $values = [
```

Comment:
This strategy of "set one timestamp and null all the rest" is not quite flexible enough, I think.  Some statuses, like PLACED_CASUAL, require having two set and one null.  Also, QUALIFIED_WITHDREW requires setting suspended_at, too.

Reply:
Huh, that seems confusing to me :sweat_smile: If final decision is getting set at the same time as placed than why differentiate them? If they are always the same, could we not just say they were placed on the final decision date? :thinking:

Reply:
Yeah, this is the mismatch between the two status systems.  When manually changing the status it's one action but with the (preferred) state machine flow, qualifying with a final decision is one action and then placing is a second action.

Reply:
Alright, updated the command to set multiple timestamps as discussed.

[allow setting multiple timestamps](https://github.com/GCTC-NTGC/gc-digital-talent/pull/12734/commits/d061155b50ebf89aee02b22eef0609e86a31196f)

Reply:
Qualified withdrew also sets `suspended_at` in the spreadsheet.  That one is a little different since that timestamp never gets nulled out in a status change, only set.

Reply:
That could be covered by:

https://github.com/GCTC-NTGC/gc-digital-talent/blob/1dd4341ffe8ab06ec30853997eab6d4768d7dca0/api/app/GraphQL/Mutations/UpdatePoolCandidateStatus.php#L53-L55

Reply:
Oh, it's underneath, sorry.  :+1: